ledger/complete/ledger.go

```diff
@@ -155,15 +155,21 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, err error) {
 	l.metrics.UpdateCount()
 	l.metrics.UpdateValuesNumber(uint64(len(trieUpdate.Paths)))
 
-	err = l.wal.RecordUpdate(trieUpdate)
-	if err != nil {
-		return nil, fmt.Errorf("cannot update state, error while writing LedgerWAL: %w", err)
-	}
+	walChan := make(chan error)
+
+	go func() {
+		walChan <- l.wal.RecordUpdate(trieUpdate)
```

Comment:
If there are blocks on different forks executed concurrently and encountered error while updating the wal, then there might be some incomplete updates to wal, right? would that be any problom, or we will be able to clean up and recover automatically on a restart?

Do we crash the node on any of the errors in updating the wal and updating the forest?

Reply:
> and encountered error while updating the wal, then there might be some incomplete updates to wal, right?

Not sure what you mean by incomplete update - the entire WAL update either succeeds or not, underlying library also adds checksums to make sure about it. 

If there are problems with updating ledger/WAL we fail block computation - but this hasn't changed. It doesn't stop EN but probably next block computation won't be able to start.